File: src/codes/Users_Manual/link/visualizer_mod.cpp
//~#include<autd3.hpp>
#include "autd3/link/visualizer.hpp"

//~int main() {
auto autd = autd3::ControllerBuilder()
                .add_device(autd3::AUTD3(autd3::Vector3::Zero()))
                .open(autd3::link::Visualizer::builder());

autd3::modulation::Sine m(150 * autd3::Hz);
autd.send(m);

autd3::link::PlotConfig config;
config.fname = "mod.png";
autd.link().plot_modulation(config, autd3::Segment::S0);
//~return 0; }

File: src/codes/Users_Manual/link/visualizer_mod.rs
# extern crate autd3;
# extern crate tokio;
# extern crate autd3_link_visualizer;
# use autd3::prelude::*;
# use autd3_link_visualizer::{Visualizer, PlotConfig};
# use std::path::Path;
# #[allow(unused_variables)]
# #[tokio::main]
# async fn main() -> Result<(), Box<dyn std::error::Error>> {
let mut autd = Controller::builder()
    .add_device(AUTD3::new(Vector3::zeros()))
    .open(Visualizer::builder()).await?;

let m = Sine::new(150. * Hz);
autd.send(m).await?;

autd.link.plot_modulation(
    PlotConfig {
        fname: Path::new("mod.png").into(),
        ..PlotConfig::default()
    },
    Segment::S0
)?;
# autd.close().await?;
# Ok(())
# }
